core/src/test/java/nl/nn/adapterframework/util/SizeLimitedVectorTest.java

```diff
@@ -0,0 +1,41 @@
+package nl.nn.adapterframework.util;
+
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * SizeLimitedVector Tester.
+ *
+ * @author <Sina Sen>
+
+ */
+public class SizeLimitedVectorTest {
+
+
+    /**
+     *
+     * Method: add(Object o)
+     *
+     */
+    @Test
+    public void testAdd() throws Exception {
+        SizeLimitedVector slv = new SizeLimitedVector(10);
+        slv.add("testString");
+        assertEquals(slv.getMaxSize(), 10);
+        assertEquals(slv.get(0), "testString");
+    }
+
+    /**
+     *
+     * Method: setMaxSize(int maxSize)
+     *
+     */
+    @Test
+    public void testSetMaxSize() throws Exception {
+        SizeLimitedVector slv = new SizeLimitedVector(10);
+        slv.setMaxSize(14);
+        assertEquals(slv.getMaxSize(), 14);
+    }
+
+
+}
```

Comment:
Try to also test the function of this class. set a limit, add limit +1 messages to the vector.
Test the size
Test that the first item has been removed

Reply:
If I understood what you meant correctly, now it is tested